### PR TITLE
Main repo lessons added for curriclum PR: #25260

### PR DESCRIPTION
**Pull request by JaeGif**
## Because
The CSS foundations lesson is exceptionally large compared to other lessons, containing a huge amount of information. To make it more digestible and easier to track progress through the lesson, it has been broken up into smaller lessons covering the major topics of the original lesson.

## This PR

- Removed CSS Foundations
- Added Intro to HTML and CSS,
- Added Intro to CSS,
- Added The Cascade

## Issue

Closes https://github.com/TheOdinProject/curriculum/issues/24064 by adding curriculum changes to main top site.

## Additional Information
[curriculum changes PR](https://github.com/TheOdinProject/curriculum/pull/25260)

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behavior
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] I have verified all tests and linters pass after making these changes.
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If applicable, this PR includes new or updated automated tests

### CODE REVIEW

File: db/fixtures/lessons/foundation_lessons.rb [hunk 1]
```diff
@@ -244,12 +244,23 @@ def foundation_lessons
       accepts_submission: true,
       has_live_preview: true,
     },
-    'CSS Foundations' => {
-      title: 'CSS Foundations',
-      description: 'Make your HTML prettier with CSS.',
-      is_project: false,
-      github_path: '/foundations/html_css/css-foundations.md',
-      identifier_uuid: 'ea39c3c3-1699-4fae-8cce-df9746480ff2',
+    'Intro to HTML and CSS' => {
```
Comment (KevinMulhern) [inline db/fixtures/lessons/foundation_lessons.rb:247]: Can we remove this lesson from here please? This lesson is the introductory lesson in the previous [HTML Foundations](https://www.theodinproject.com/paths/foundations/courses/foundations#html-foundations) section.

File: db/fixtures/paths/foundations/seed.rb [hunk 2]
```diff
@@ -101,7 +101,9 @@
   section.identifier_uuid = 'd2476929-d71b-4b25-969b-8f7da6a40c94'
 
   section.add_lessons(
-    foundation_lessons.fetch('CSS Foundations'),
+    foundation_lessons.fetch('Intro to HTML and CSS'),
```
Comment (KevinMulhern) [inline db/fixtures/paths/foundations/seed.rb:104]: Same here please, RE: this lesson comes earlier in the course.

```suggestion
```

**Review by KevinMulhern — CHANGES_REQUESTED** [review]
Hey @JaeGif this is looks great at a glance. We'll need [the curriculum PR](https://github.com/TheOdinProject/curriculum/pull/25260) merged to test these changes properly. Before we can merge that, we need the merge conflict resolved. If you need any help with the merge conflict, please let us know.

The only feedback I have for this PR so far is we just need to remove the erroneous lesson.

File: db/fixtures/paths/foundations/seed.rb (hunk 2 above)
Comment (JaeGif) [reply db/fixtures/paths/foundations/seed.rb]: Got it

**Review by KevinMulhern — APPROVED** [review]
This looks great on the review app. Thanks for all the work you've put into this @JaeGif, it's going to be a big QOL upgrade for the Foundations course 🎉

**Pull request merged**